contrib/scrooge/src/python/pants/contrib/scrooge/tasks/scrooge_gen.py

```diff
@@ -135,20 +135,22 @@ def execute_codegen(self, target, target_workdir):
     self._must_have_sources(target)
 
     namespace_map = self._thrift_defaults.namespace_map(target)
+    compiler_args = self._thrift_defaults.compiler_args(target)
 
     partial_cmd = self.PartialCmd(
         language=self._validate_language(target),
         rpc_style=self._validate_rpc_style(target),
         namespace_map=tuple(sorted(namespace_map.items())) if namespace_map else (),
         default_java_namespace=self._thrift_defaults.default_java_namespace(target),
-        include_paths=target.include_paths)
+        include_paths=target.include_paths,
+        compiler_args=compiler_args)
 
     self.gen(partial_cmd, target, target_workdir)
 
   def gen(self, partial_cmd, target, target_workdir):
     import_paths, _ = calculate_compile_sources([target], self.is_gentarget)
 
-    args = []
+    args = partial_cmd.compiler_args
```

Comment:
In order to not mutate these, I think you need to copy them... probably with `args = list(partial_cmd.compiler_args)`.